avaframe/com1DFAPy/com1DFA.py

```diff
@@ -203,15 +206,33 @@ def getSimulation(cfg, rel):
             relDict['d0'][k] = float(relDict['d0'][k])
 
     log.info('Release area scenario: %s - perform simulations' % (relName))
-    if entRes:
-        # Possibility to run only entrainment resistance or also with null
-        if onlyEntrRes:
-            cuSim = [simName + '_entres_dfa']
-        else:
-            cuSim = [simName + '_null_dfa', simName + '_entres_dfa']
-    else:
-        # Initialise CreateSimulations cint file and set parameters
-        cuSim = [simName + '_null_dfa']
+
+    # define simulation type
+    cuSim = []
+    for simType in simTypeList:
+        if simType == 'ent':
+            if entResInfo['flagEnt'] == 'Yes':
+                cuSim.append(simName + '_ent_dfa')
+            else:
+                log.warning('No entrainment area available for ent simulation')
+        elif simType == 'entres':
+            if entResInfo['flagEnt'] == 'Yes' and entResInfo['flagRes'] == 'Yes':
+                cuSim.append(simName + '_entres_dfa')
+            else:
+                if entResInfo['flagEnt'] == 'Yes' and entResInfo['flagRes'] == 'No':
+                    log.warning('No resistance area available for entres simulation')
+                elif entResInfo['flagEnt'] == 'No' and entResInfo['flagRes'] == 'Yes':
+                    log.warning('No entrainment area available for entres simulation')
+                elif entResInfo['flagEnt'] == 'No' and entResInfo['flagRes'] == 'No':
+                    log.warning('No resistance and no entrainment area available for entres simulation')
+        elif simType == 'res':
+            if entResInfo['flagRes'] == 'Yes':
+                cuSim.append(simName + '_res_dfa')
+            else:
+                log.warning('No resistance area available for res simulation')
+        elif simType == 'null':
+            cuSim.append(simName + '_null_dfa')
+
```

Comment:
Mh, can we simplify this somehow? The way it is right now: it is hard to read and understand what's going on...